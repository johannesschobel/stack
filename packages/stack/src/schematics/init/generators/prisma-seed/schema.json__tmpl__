{
  "cli": "nx",
  "id": "prisma-seed",
  "type": "object",
  "properties": {
    "confirm": {
      "type": "boolean",
      "description": "Library confirm",
      "x-prompt": "Are you sure you want to delete and seed the database?"
    }
  },
  "required": ["confirm"]
}
