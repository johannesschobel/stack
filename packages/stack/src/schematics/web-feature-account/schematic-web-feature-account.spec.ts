import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { createEmptyWorkspace } from '@nrwl/workspace/testing'
import { join } from 'path'

import { WebFeatureAccountSchematicSchema } from './schema'

describe('web-feature-account schematic', () => {
  let appTree: Tree
  const options: WebFeatureAccountSchematicSchema = { appName: 'test' }

  const testRunner = new SchematicTestRunner('@nxpm/web-feature-account', join(__dirname, '../../../collection.json'))

  beforeEach(() => {
    appTree = createEmptyWorkspace(Tree.empty())
  })

  it('should run successfully', async () => {
    await expect(
      testRunner.runSchematicAsync('web-feature-account', options, appTree).toPromise(),
    ).resolves.not.toThrowError()

    expect(
      appTree.read('libs/test/account/feature/src/lib/test-account-feature.module.ts').toString(),
    ).toMatchSnapshot()
  })
})
