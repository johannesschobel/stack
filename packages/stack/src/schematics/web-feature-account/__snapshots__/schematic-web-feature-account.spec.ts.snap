// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web-feature-account schematic should run successfully 1`] = `
"import { CommonModule } from '@angular/common'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { TestUiSidebarPageModule } from '@nxpm/test/ui/sidebar-page'
import { TestAccountFeatureComponent } from './test-account-feature.component'

@NgModule({
  declarations: [TestAccountFeatureComponent],
  imports: [
    CommonModule,
    RouterModule.forChild([
      {
        path: '',
        component: TestAccountFeatureComponent,
        children: [
          { path: '', pathMatch: 'full', redirectTo: 'profile' },
          {
            path: 'profile',
            loadChildren: () => import('./account-profile/account-profile.module').then((m) => m.AccountProfileModule),
          },
        ],
      },
    ]),
    TestUiSidebarPageModule,
  ],
})
export class TestAccountFeatureModule {}
"
`;
